Remove NLopt's dependency on its internal luksan algorithm library.

That library is licensed under LGPL-2.1+ but the rest of NLopt is
licensed under MIT or similar notice-only licenses, and we really
don't want to distribute this code using dynamic linking.

--- src/api/optimize.c.orig
+++ src/api/optimize.c
@@ -40,7 +40,9 @@
 
 #include "cdirect.h"
 
+#if 0
 #include "luksan.h"
+#endif
 
 #include "crs.h"
 
@@ -573,18 +575,18 @@
             return praxis_(nlopt_get_param(opt, "t0_tol", 0.0), DBL_EPSILON, step, ni, x, f_bound, opt, &stop, minf);
         }
 
+#if 0
     case NLOPT_LD_LBFGS:
         return luksan_plis(ni, f, f_data, lb, ub, x, minf, &stop, opt->vector_storage);
-
     case NLOPT_LD_VAR1:
     case NLOPT_LD_VAR2:
         return luksan_plip(ni, f, f_data, lb, ub, x, minf, &stop, opt->vector_storage, algorithm == NLOPT_LD_VAR1 ? 1 : 2);
-
     case NLOPT_LD_TNEWTON:
     case NLOPT_LD_TNEWTON_RESTART:
     case NLOPT_LD_TNEWTON_PRECOND:
     case NLOPT_LD_TNEWTON_PRECOND_RESTART:
         return luksan_pnet(ni, f, f_data, lb, ub, x, minf, &stop, opt->vector_storage, 1 + (algorithm - NLOPT_LD_TNEWTON) % 2, 1 + (algorithm - NLOPT_LD_TNEWTON) / 2);
+#endif
 
     case NLOPT_GN_CRS2_LM:
         if (!finite_domain(n, lb, ub))
